### PR TITLE
Fix #203: Octal constants not handled correctly

### LINKED ISSUE #203: Octal constants not handled correctly
Using DStep v0.2.3-16-g1308991 compiled on Debian Sid with LDC 1.11.0 and LLVM 6.0, I generated a module for fontconfig.h (version 2.13.0). Sadly:
```
generated/fontconfig.d(932): Error: octal literals 0101 are no longer supported, use std.conv.octal!101 instead
generated/fontconfig.d(932): Error: octal literals 0132 are no longer supported, use std.conv.octal!132 instead
generated/fontconfig.d(937): Error: octal literals 0141 are no longer supported, use std.conv.octal!141 instead
generated/fontconfig.d(937): Error: octal literals 0172 are no longer supported, use std.conv.octal!172 instead
generated/fontconfig.d(942): Error: octal literals 0101 are no longer supported, use std.conv.octal!101 instead
generated/fontconfig.d(942): Error: octal literals 0141 are no longer supported, use std.conv.octal!141 instead
```

### PR DESCRIPTION
**Pull request by ciechowoj**

### CODE REVIEW

File: tests/unit/UnitTests.d [hunk 1]
```diff
@@ -932,3 +932,21 @@ void bar (const(foo_t)* foo);
...
+{
+    assertTranslates(
+    q"C
+#define some_octal_number 0101
+#define zero_is_not_octal 0
+#define double_zero_is_octal 00
+C", q"D
+extern (C):
+
+enum some_octal_number = octal!101;
```
Comment (jacob-carlborg) [inline tests/unit/UnitTests.d]: Why is there no import for `octal` in this test?

Comment (ciechowoj) [reply tests/unit/UnitTests.d]: Good point!

Comment (ciechowoj) [reply tests/unit/UnitTests.d]: Fixed.

File: dstep/translator/MacroDefinition.d [hunk 2]
```diff
@@ -100,6 +100,30 @@ struct ExpressionContext
...
+{
+    import std.algorithm;
+    import std.ascii;
+
+    if (literal.spelling.length > 1 &&
+        all!isDigit(literal.spelling) &&
+        literal.spelling.front == '0')
+    {
+        (*context.imports).add("std.conv : octal");
+        return "octal!" ~ literal.spelling[0..$-1].stripLeft('0') ~ literal.spelling[$-1];
```
Comment (jacob-carlborg) [inline dstep/translator/MacroDefinition.d]: Doesn't look like unsigned and long versions of the literals are handled properly, i.e. `0700u` and `0700L`.

Comment (ciechowoj) [reply dstep/translator/MacroDefinition.d]: Fixed.

**Comment by ciechowoj** [comment]
@jacob-carlborg 
What about merging it to master?

**Pull request merged**